src/main/java/step3/CarRacing.java

```diff
@@ -0,0 +1,52 @@
+package step3;
+
+import java.util.Random;
+
+public class CarRacing {
+	private static final int CAN_MOVE_NUM = 4;
+	private static final int RANDOM_LIMIT = 9;
+	private static final String MOVE_MARK = "-";
+	private static final Random random = new Random();
+
+	public static void main(String[] args) {
+		int carCount = InputView.requestHowManyCar();
+		int times = InputView.requestHowManyTimes();
+
+		StringBuilder[] cars = getNewCars(carCount);
+		StringBuilder[] raceResult = new StringBuilder[times];
+
+		for (int i = 0; i < times; i++) {
+			raceResult[i] = moveForward(cars);
+		}
+
+		ResultView.printResult(raceResult);
+	}
+
+	public static StringBuilder[] getNewCars(int carCount) {
+		StringBuilder[] cars = new StringBuilder[carCount];
+		for (int i = 0; i < cars.length; i++) {
+			cars[i] = new StringBuilder();
+		}
+		return cars;
+	}
+
+	private static StringBuilder moveForward(StringBuilder[] cars) {
+		StringBuilder roundResult = new StringBuilder();
+		for (int i=0; i < cars.length; i++) {
+			cars[i] = canMoveForward(cars[i]);
+			roundResult.append(cars[i]).append("\n");
+		}
+		return roundResult;
+	}
+
+	public static StringBuilder canMoveForward(StringBuilder car) {
+		if(isMoveNum(random.nextInt(RANDOM_LIMIT))) {
```

Comment:
현재 미션의 요구사항은 전진하는 조건은 0에서 9 사이에서 random 값을 구한 후 random 값이 4이상일 경우인데요
현재 메서드 내부에서 Random값을 생성한다면 이동했는지, 멈춰있는지에 테스트를 하지 못할 것 같아요.
테스트를 못하는 구조는 리팩토링이 필요한 신호입니다.

랜덤과 같은 무작위 테스트의 문제점은 예상되는 값이 없기에 테스트를 할 수가 없습니다. 
그렇다면 테스트를 해야하는 부분은 랜덤값 생성에 대한 테스트가 아니라 랜덤 요소가 들어간 로직이 되지 않을까 싶습니다.
랜덤값을 외부에서 받도록 하여 테스트 가능한 코드와 테스트하기 힘든 부분을 분리해보시면 좋을 것 같습니다 :)

```suggestion
		if (isMoveNum(random.nextInt(RANDOM_LIMIT))) {
```